glusterd2/xlator/options/options.go

```diff
@@ -254,21 +260,37 @@ func ValidatePriorityOrSize(o *Option, val string) error {
 
 // ValidateSizeList validates if the option is a valid size list
 func ValidateSizeList(o *Option, val string) error {
-	return ErrInvalidArg
+	var sizeinbytes int64
+	slist := strings.Split(val, ",")
+	for _, el := range slist {
+		l := len(el)
+		if strings.ContainsRune(el, 'B') || strings.ContainsRune(el, 'b') {
```

Comment:
You should instead check if `el` ends with `B` or `b`. The `ContainsRune` checks if it's present in the string anywhere. You can use [this](https://golang.org/pkg/strings/#HasSuffix) instead.